inginious/frontend/pages/tasks.py

```diff
@@ -261,15 +240,97 @@ def POST_AUTH(self, courseid, taskid, isLTI):
+            else:
+                text = _("<b>There are {} tasks in front of you in the waiting queue.</b>").format(nb_tasks_before)
+
+            return json.dumps({'status': "waiting", 'text': text})
+
+        tojson = {
+            'status': data['status'],
+            'result': data.get('result', 'crash'),
+            'id': str(data["_id"]),
+            'submitted_on': str(data['submitted_on']),
+            'grade': str(data.get("grade", 0.0)),
+            'replace': replace and not reloading  # Replace the evaluated submission
+        }
+
+        if "text" in data:
+            tojson["text"] = data["text"]
+        if "problems" in data:
+            tojson["problems"] = data["problems"]
+
+        if debug:
+            tojson["debug"] = data
+
+        if data['status'] == 'waiting':
+            tojson["text"] = _("<b>Your submission has been sent...</b>")
+        elif data["result"] == "failed":
+            tojson["text"] = _("There are some errors in your answer. Your score is {score}%.").format(score=data["grade"])
+        elif data["result"] == "success":
+            tojson["text"] = _("Your answer passed the tests! Your score is {score}%.").format(score=data["grade"])
+        elif data["result"] == "timeout":
+            tojson["text"] = _("Your submission timed out. Your score is {score}%.").format(score=data["grade"])
+        elif data["result"] == "overflow":
+            tojson["text"] = _("Your submission made an overflow. Your score is {score}%.").format(score=data["grade"])
+        elif data["result"] == "killed":
+            tojson["text"] = _("Your submission was killed.")
+        else:
+            tojson["text"] = _("An internal error occurred. Please retry later. "
+                               "If the error persists, send an email to the course administrator.")
+
+        tojson["text"] = "<b>" + tojson["text"] + " " + _("[Submission #{submissionid}]").format(submissionid=data["_id"]) + "</b>" + data.get("text", "")
+
+        if reloading:
+            # Set status='ok' because we are reloading an old submission.
+            tojson["status"] = 'ok'
+            # And also include input
+            tojson["input"] = data.get('input', {})
+
+        return json.dumps(tojson, default=str)
+
+    def list_multiple_multiple_choices_and_files(self, task):
+        """ List problems in task that expect and array as input """
+        output = {}
+        for problem in task.get_problems():
+            if isinstance(problem, MultipleChoiceProblem) and problem.allow_multiple():
+                output[problem.get_id()] = []
+            elif isinstance(problem, BasicCodeProblem):
+                for box in problem.get_boxes():
+                    if isinstance(box, FileBox):
+                        output[box.get_complete_id()] = {}
+        return output
+
 
-class TaskPageStaticDownload(INGIniousAuthPage):
+class TaskPageStaticDownload(INGIniousPage):
```

Comment:
Do not forget to add a warning in doc about the full availability of public files.